components/timezone/src/ixdtf.rs

```diff
@@ -78,116 +78,214 @@ impl UtcOffset {
-        }
-    }
-
-    fn try_from_utc_offset_record(record: &UTCOffsetRecord) -> Result<Self, ParseError> {
+            _ => return Err(ParseError::MismatchedTimeZoneFields),
+        };
+        let offset = UtcOffset::try_from_utc_offset_record(offset)?;
+        let iso = DateTime::<Iso>::try_new_iso_datetime(
+            date.year,
+            date.month,
+            date.day,
+            time.hour,
+            time.minute,
+            time.second,
+        )?;
         Ok(Self {
             time_zone_id: crate::TimeZoneBcp47Id::unknown(),
-            offset: Some(UtcOffset::try_from_utc_offset_record(record)?),
-            zone_variant: Some(ZoneVariant::standard()),
-            local_time: Some((Date::unix_epoch(), Time::midnight())),
+            offset: Some(offset),
+            zone_variant: ZoneVariant::standard(),
+            local_time: (iso.date, iso.time),
         })
     }
+}
 
-    fn try_from_time_zone_record(
-        record: &TimeZoneRecord<'_>,
-        offset: Option<UtcOffset>,
-        date: &Option<DateRecord>,
-        time: &Option<TimeRecord>,
+impl TimeZoneInfo<models::AtTime> {
+    fn try_location_only_from_ixdtf_record(
+        ixdtf_record: &IxdtfParseRecord,
     ) -> Result<Self, ParseError> {
-        match record {
-            TimeZoneRecord::Name(iana_identifier) => {
-                let mapper = TimeZoneIdMapper::new();
-                let time_zone_id = mapper.as_borrowed().iana_bytes_to_bcp47(iana_identifier);
-
-                let mut zone_variant = None;
-                let mut local_time = None;
-                if let (Some(date), Some(time)) = (date, time) {
-                    let iso = DateTime::<Iso>::try_new_iso_datetime(
-                        date.year,
-                        date.month,
-                        date.day,
-                        time.hour,
-                        time.minute,
-                        time.second,
-                    )?;
-                    if let Some(offset) = offset {
-                        zone_variant = Some(
-                            if let Some((std_offset, dst_offset)) = ZoneOffsetCalculator::new()
-                                .compute_offsets_from_time_zone(time_zone_id, &iso)
-                            {
-                                if offset == std_offset {
-                                    ZoneVariant::standard()
-                                } else if Some(offset) == dst_offset {
-                                    ZoneVariant::daylight()
-                                } else {
-                                    return Err(ParseError::InvalidOffsetError);
-                                }
-                            } else {
-                                debug_assert_eq!(time_zone_id.0.as_str(), "unk");
-                                ZoneVariant::standard()
-                            },
-                        );
+        let IxdtfParseRecord {
+            offset: None,
```

Comment:
phew, not sure I want to make too much information an error. Imagine if you want to start sending more details across the network to change time zone format, if this errors you have to upgrade client and server at exactly the same time.

Reply:
I think being strict about the expected fields is important. But, we can continue to iterate on exactly the shape of these functions and the strictness. In the mean time, I added more comments.

Reply:
Imagine if you're calling an API that only guarantees that it sends the offset, but doesn't guarantee that it *only* sends the offset.

Reply:
Perhaps that use case should be another constructor / parse function that sets the time zone to `unk` if it is not present. It seems bad to _throw away_ the info, though.

Reply:
There's currently not constructor here that does that.

Reply:
(Do you want it now or can we add it later when we make a polished IXDTF parser?)

Reply:
You're saying this is a workaround to the issue I'm describing. You can address this issue either by removing the error path here, or by adding a more lenient constructor.

Reply:
This PR has

- `try_offset_only_from_str` = strict offset only parsing
- `try_location_only_from_str` = strict location only parsing
- `try_from_str` = strict offset+location parsing

I can add:

- `try_loose_from_str` = missing offset parses to `None`; missing location parses to `unk`

Reply:
This does not address the issue here. If I only need location info, it should not be an error for there to be offset info. It should be an error for there not to be location info.

Reply:
Yes, it should be an error. The constructor you choose asserts the shape of the string.

Can we please not make this block the PR.

Reply:
No this is important. Look at serde json, it does not complain when an object contains a key it doesn't know, it only complains when it's missing a key.

Reply:
ok can I make it `try_strict_from_str` and then `try_from_str` is the loose one, as it is now

Reply:
yes

Reply:
There are 4 functions which I think cover all of the use cases. This is how I have named them for now:

- try_offset_only_iso_from_str (strict, returns `UtcOffset`)
- try_location_only_iso_from_str (strict, returns `TimeZoneInfo<AtTime>`)
- try_loose_iso_from_str (loose, returns `TimeZoneInfo<AtTime>`)
- try_iso_from_str (strict, returns `TimeZoneInfo<Full>`)

There is only one "loose" function because it is only possible for a loose function to return `TimeZoneInfo<AtTime>`, not `TimeZoneInfo<Full>`.

I do not see a use case for a "loose" function returning `UtcOffset` because if you want things loose, just get a `TimeZoneInfo<AtTime>` which is more general and fully featured.